libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -46,9 +50,8 @@ void DagBlockPacketHandler::process(const PacketData &packet_data, const std::sh
       return;
     }
 
+    trx_mgr_->insertBroadcastedTransactions(new_transactions);
     if (auto status = syncing_handler_->checkDagBlockValidation(block); !status.first) {
-      trx_mgr_->insertBroadcastedTransactions(new_transactions);
```

Comment:
So it might be little bit too "hot" hotfix :) I checked and with this approach we would process txs twice or at least try to process it twice. I would keep insertBroadcastedTransactions here and also put it maybe inside onNewBlockReceived -> insertBroadcastedBlockWithTransactions. Then we dont need and probably should not to save txs into the dag unverified queue.

Reply:
another point is that I think we might get into this error also inside dag_sync_packet_handler -> process() function where we are doing basically the same stuff as here but we dont insert txs in case of failed dag block validation. Not sure if that is valid scenario though...